Classes/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Pelago\Emogrifier\HtmlProcessor;
+
+/**
+ * Base class for HTML processor that e.g., can remove, add or modify nodes or attributes.
+ *
+ * @author Oliver Klee <github@oliverklee.de>
+ */
+abstract class AbstractHtmlProcessor
+{
+    /**
+     * @var string
+     */
+    const DEFAULT_DOCUMENT_TYPE = '<!DOCTYPE html>';
+
+    /**
+     * @var string
+     */
+    const CONTENT_TYPE_META_TAG = '<meta http-equiv="Content-Type" content="text/html; charset=utf-8">';
+
+    /**
+     * @var \DOMDocument
+     */
+    private $xmlDocument = null;
```

Comment:
I think this property should be protected instead of private, so child classes have access to it. Or getter method must be implemented.

Reply:
Will do.